src/spikeinterface/core/sparsity.py

```diff
@@ -204,7 +205,11 @@ def are_waveforms_dense(self, waveforms: np.ndarray) -> bool:
     def are_waveforms_sparse(self, waveforms: np.ndarray, unit_id: str | int) -> bool:
         non_zero_indices = self.unit_id_to_channel_indices[unit_id]
         num_active_channels = len(non_zero_indices)
-        return waveforms.shape[-1] == num_active_channels
+
+        # If any channel is non-zero outside of the active channels, then the waveforms are not sparse
+        excess_zeros = waveforms[..., num_active_channels:].sum()
+
+        return int(excess_zeros) == 0
```

Comment:
Not sure to be happy with this.
This is computationaly costy. For me the shape is enough.

Reply:
ok for now. If this becomes problematic later we'll fix it.

NOTE: this assumes that the extra channels are zero-padded!

Reply:
Added a boolean to deactivate this.